src/reader.rs

```diff
@@ -636,7 +608,10 @@ impl<R: Read + Seek> VdbReader<R> {
             }
             gd.meta_data = Self::read_metadata(reader)?;
 
-            result.push(gd);
+            assert!(
+                result.insert(name, gd).is_none(),
+                "Name clash found in VDB grids"
```

Comment:
```suggestion
                "Grid named {name} already exists"
```

Or something else that includes `{name}` :)